src/Compilers/Server/ServerShared/DiagnosticListener.cs

```diff
@@ -16,10 +16,15 @@ internal interface IDiagnosticListener
         void UpdateKeepAlive(TimeSpan timeSpan);
 
         /// <summary>
+        /// Called when a connection to the server occurs.
+        /// </summary>
+        void Connection();
```

Comment:
Btw that doesn't mean they should be `event`, just that I'm still hazy on what best practices for `event` vs. not-`event` are.